### PR TITLE
[Security] Add Client-Side Math Expression Evaluation for TSVB

### PR DESCRIPTION
**Pull request by Maosaic**
## Summary

This PR implements client-side math expression evaluation for TSVB (Time Series Visual Builder) visualizations, addressing security and scalability concerns identified in [RFC #11081](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/11081).

**What:** Adds a new `/api/metrics/vis/data-raw` endpoint that returns individual metric values without evaluating math expressions, then evaluates those expressions client-side in the browser.

**Why:** Reduces server-side attack surface by moving arbitrary expression evaluation out of Node.js, improves scalability by offloading computation to clients, and leverages browser security sandboxing.

**How:** Creates parallel implementation that coexists with existing server-side evaluation, controllable via UI setting.

### Current Flow (unchanged)
```
Client → /api/metrics/vis/data → Server evaluates math → Returns computed results
```

### New Flow (opt-in via UI setting)
```
Client → /api/metrics/vis/data-raw → Server skips math → Returns raw metrics
       → Client evaluates math → Renders visualization
```

### Server-Side (5 new files)

Created new endpoint pipeline that mirrors existing `/api/metrics/vis/data` but excludes math evaluation:

1. **`server/routes/vis_raw.ts`** - New route handler for `/api/metrics/vis/data-raw`
2. **`server/lib/get_vis_data_raw.ts`** - Vis data handler (no math)
3. **`server/lib/vis_data/get_series_data_raw.js`** - Series data handler (no math)
4. **`server/lib/vis_data/series/handle_response_body_raw.js`** - Response handler (no math)
5. **`server/lib/vis_data/response_processors/series/index_raw.js`** - Processor list excluding `mathAgg`

### Client-Side (2 new modules)

1. **`public/lib/mathjs_wrapper.js`** - Security-hardened Math.js wrapper
   - Configures Math.js with same security restrictions as server
   - Blocks dangerous functions (import, parse, evaluate, createUnit, etc.)
   - Provides simple `evaluate(expression, scope)` interface

2. **`public/lib/process_math_series.js`** - TSVB math processor
   - Processes raw response from `/api/metrics/vis/data-raw`
   - Builds evaluation context matching server exactly:
     - `params.a`, `params.b`, etc. (variable values)
     - `params._index` (current position in series)
     - `params._timestamp` (current timestamp)
     - `params._all` (entire series data for advanced expressions)
     - `params._interval` (bucket size in milliseconds)
   - Handles edge cases: division by zero → null, null values → null
   - Returns processed series ready for visualization

3. **`public/request_handler_raw.js`** - Request handler using new endpoint

### Configuration & Integration

- **`common/constants.ts`** - Added `CLIENT_SIDE_EVALUATION_SETTING`
- **`server/ui_settings.ts`** - Added UI setting: "TSVB client-side math evaluation"
  - Category: Visualization
  - Default: `true` (client-side enabled)
  - Allows users to toggle between client/server evaluation
- **`public/metrics_type.ts`** - Request handler wrapper reads setting at runtime
- **`public/metrics_fn.ts`** - Expression function wrapper reads setting at runtime

### Backwards Compatibility

✅ **Fully backwards compatible:**

- Existing `/api/metrics/vis/data` endpoint unchanged
- Existing server-side evaluation unchanged
- Saved visualizations work unchanged
- Default behavior is client-side (but configurable)
- Users can toggle back to server-side via UI setting
- No schema changes required
- No migration needed

## Screenshot

Screenshot 2026-01-05 at 3 54 06 PM

Screenshot 2026-01-05 at 3 56 11 PM

## Changelog

- feat: Add Client-Side Math Expression Evaluation for TSVB

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

## Summary by CodeRabbit

* **New Features**
  * Optional client-side math expression evaluation for time series visualizations with secure expression handling.
  * New UI toggle to enable/disable client-side evaluation.

* **Bug Fixes**
  * Improved null handling in math evaluation and stricter error propagation to avoid silent failures.

* **Tests**
  * Extensive test coverage for client-side math evaluation, edge cases, and security restrictions.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/plugins/vis_type_timeseries/public/lib/process_math_series.js [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+          ...params,
+          _index: index,
+          _timestamp: ts,
+          _all: all,
+          _interval: bucketSize * 1000,
+        },
+      });
+
+      // Handle object results (flatten and return last value)
+      if (typeof result === 'object') {
```
Comment (ruanyl) [inline src/plugins/vis_type_timeseries/public/lib/process_math_series.js]: Nit: should it check `result !== null` before `typeof`?

File: src/plugins/vis_type_timeseries/public/request_handler_raw.js
Comment (ruanyl) [inline src/plugins/vis_type_timeseries/public/request_handler_raw.js:1]: Curious why not use `.ts`?

**Review by ruanyl — DISMISSED** [review]

File: src/plugins/vis_type_timeseries/public/request_handler_raw.js
Comment (Maosaic) [reply src/plugins/vis_type_timeseries/public/request_handler_raw.js:1]: I'm following src/plugins/vis_type_timeseries/public/request_handler.js

**Review by ruanyl — APPROVED** [review]
